### ISSUE TITLE
Migrate version-dependent documentation to Read The Docs

### ISSUE THREAD

**Issue by davidhorstmann-arm** [Mbed-TLS/mbedtls#4848]
#3643 covered the task of migrating the documentation to Read The Docs. Due to some issues with the Sphinx Breathe plugin, only the version-independent documentation in ARMmbed/mbedtls-docs was successfully migrated. This issue tracks the migration of the version-dependent documentation in ARMmbed/mbedtls, most importantly the doxygen API documentation and the user facing documents in the `docs/` directory.

The task is completed when:
* The correct Sphinx configuration is added to ARMmbed/mbedtls to build the version-dependent docs
* The version-dependent docs are added as a subproject of the Read The Docs project

**Comment by yanesca** [comment]
Related task: https://github.com/ARMmbed/mbedtls/issues/4862

**Comment by davidhorstmann-arm** [comment]
When trying to use the Sphinx Breathe plugin to ingest our doxygen documentation I encountered doxygen/doxygen#8706, which should be fixed by doxygen/doxygen#9207. This means we will be able to use Breathe with the versioned documentation once the fix is merged.

**Comment by gilles-peskine-arm** [comment]
@davidhorstmann-arm Can we work around this by tweaking our doxygen code, or is it inherent in using Breathe or too invasive?

**Comment by davidhorstmann-arm** [comment]
Yes, the workaround is very easy - the issue is that in `bignum.h` we have:
```
 * \param rounds   The number of bases to perform the Miller-Rabin primality
 *                 test for. The probability of returning 0 on a composite is
 *                 at most 2<sup>-2*\p rounds</sup>.
```
We can work around it by adding a space before the closing `</sup>` tag, like so:
`2<sup>-2*\p rounds</sup>` -> `2<sup>-2*\p rounds </sup>`

My previous comment was mainly for documentary purposes, and to link to the doxygen issue in case future people encounter it.

**Comment by gilles-peskine-arm** [comment]
This encompasses both Doxygen documentation (which will resolve https://github.com/Mbed-TLS/mbedtls/issues/5955) and non-Doxygen documentation (e.g. `getting_started.md`, which lacks visibility now that readthedocs is the primary interface to mbedtls documentation).

**Comment by davidhorstmann-arm** [comment]
Hmm actually this was the wrong issue to close, as we only have API documentation up. I will close #5955 instead.
